Google.PowerShell/Common/GcloudWrapper.cs

```diff
@@ -0,0 +1,109 @@
+﻿// Copyright 2016 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using Newtonsoft.Json.Linq;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Threading.Tasks;
+
+namespace Google.PowerShell.Common
+{
+    /// <summary>
+    /// This class wraps the gcloud command and offers up some of its services.
+    /// </summary>
+    public static class GCloudWrapper
+    {
+        /// <summary>
+        /// Returns the global installation properties path of GoogleCloud SDK.
+        /// </summary>
+        /// <returns></returns>
+        public static async Task<string> GetInstallationPropertiesPath()
+        {
+            string gCloudInfoOutput = await GetGCloudCommandOutput("info");
+            JToken gCloudInfoJson = JObject.Parse(gCloudInfoOutput);
+
+            try
+            {
+                // SelectToken will throw NullReferenceException if the token cannot be found.
```

Comment:
Is there a way to check if the token exists? Ultimately we should avoid having the exception be thrown if we can. There is a huge perf hit and we can also provide a more accurate error message. (e.g. "Got configuration value, but couldn't find installation_properties_path."